File: 1.1/src/cz/mg/vulkan/PFNvkImportFenceFdKHR.java
package cz.mg.vulkan;

public class PFNvkImportFenceFdKHR extends VkFunctionPointer {
    public PFNvkImportFenceFdKHR() {
    }

    protected PFNvkImportFenceFdKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkImportFenceFdKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkImportFenceFdKHR(long value) {
        setValue(value);
    }

    public PFNvkImportFenceFdKHR(VkInstance instance) {
        super(instance, new VkString("vkImportFenceFdKHR"));
    }

    public void call(VkDevice device, VkImportFenceFdInfoKHR pImportFenceFdInfo, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pImportFenceFdInfo != null ? pImportFenceFdInfo.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long device, long pImportFenceFdInfo, long rval);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkImportFenceFdKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkImportFenceFdKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pImportFenceFdInfo, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkImportFenceFdKHR f = (PFN_vkImportFenceFdKHR)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkImportFenceFdInfoKHR*)jniLongToPointer(pImportFenceFdInfo))
    );
}


jint Java_cz_mg_vulkan_PFNvkImportFenceFdKHR_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pImportFenceFdInfo){
    (void)env;
    (void)jc;
    PFN_vkImportFenceFdKHR f = (PFN_vkImportFenceFdKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkImportFenceFdInfoKHR*)jniLongToPointer(pImportFenceFdInfo))
    );
}

